File: packages/typescript/src/endpoints/ai_results_export.ts
/**
 * Ai_Results_Export API Client
 */

import { AxiosInstance } from 'axios'

export class Ai_Results_ExportClient {
  private httpClient: AxiosInstance

  constructor(private baseURL: string, httpClient: AxiosInstance) {
    this.httpClient = httpClient
  }

  /**
   * GET /results
   */
  async getResults(skip?: number, limit?: number, scan_id?: string, analysis_type?: string, severity?: string, start_date?: string, end_date?: string): Promise<Dict[str, Any]> {
        const params = {'skip: skip, limit: limit, scan_id: scan_id, analysis_type: analysis_type, severity: severity, start_date: start_date, end_date: end_date'}
        const data = undefined
        return this.httpClient.get(`/results`, params)
  }
  /**
   * GET /results/export
   */
  async getResultsexport(format?: string, scan_id?: string, analysis_type?: string, start_date?: string, end_date?: string): Promise<Response> {
        const params = {'format: format, scan_id: scan_id, analysis_type: analysis_type, start_date: start_date, end_date: end_date'}
        const data = undefined
        return this.httpClient.get(`/results/export`, params)
  }
}

File: packages/python/src/endpoints/ai_results_export.py
"""
Ai_Results_Export API Client
"""

from typing import Dict, List, Optional, Any
import httpx


class Ai_Results_ExportClient:
    """Client for ai_results_export endpoints."""

    def __init__(self, base_url: str, client: Optional[httpx.AsyncClient] = None):
        self.base_url = base_url.rstrip('/')
        self.client = client

    async def _request(self, method: str, path: str, **kwargs) -> Any:
        """Make HTTP request."""
        url = self.base_url + path
        if self.client:
            response = await self.client.request(method, url, **kwargs)
            response.raise_for_status()
            return response.json()
        else:
            async with httpx.AsyncClient() as client:
                response = await client.request(method, url, **kwargs)
                response.raise_for_status()
                return response.json()


    async def get_results(self, skip: Optional[float] = None, limit: Optional[float] = None, scan_id: Optional[str] = None, analysis_type: Optional[str] = None, severity: Optional[str] = None, start_date: Optional[str] = None, end_date: Optional[str] = None):
        """GET /results"""
        url = f"/results"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)


    async def get_results_export(self, format: Optional[str] = None, scan_id: Optional[str] = None, analysis_type: Optional[str] = None, start_date: Optional[str] = None, end_date: Optional[str] = None):
        """GET /results/export"""
        url = f"/results/export"
        params = {", ".join(query_dict)}
        data = None
        return await self._request("GET", url, params=params, json=data)

